scipy/optimize/optimize.py

```diff
@@ -1866,14 +1873,18 @@ def _minimize_scalar_bounded(func, bounds, args=(),
             flag = 1
             break
 
+    if np.isnan(xf) or np.isnan(fx):
+        flag = 2
```

Comment:
Azure coverage indicates that `flag = 2` is never hit

Reply:
Unrelated, not such a good idea to have variables `xf` and `fx`, they'd be easily mixed up.

Reply:
Added tests also for the scalar minimizers.
Renaming variables should be in a different PR (if at all).